pkg/project/project_test.go

```diff
@@ -103,6 +167,63 @@ func TestCreate(t *testing.T) {
 		})
 	}
 
+	// run tests for Kubernetes (Namespace)
+	for _, tt := range tests {
+		t.Run(fmt.Sprintf(" %s with Namespace", tt.name), func(t *testing.T) {
+
+			// Fake the client with the appropriate arguments
+			client, fakeClientSet := occlient.FakeNew()
+
+			loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+			configOverrides := &clientcmd.ConfigOverrides{}
+			client.KubeConfig = clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+			fkWatch := watch.NewFake()
+
+			fakeClientSet.Kubernetes.PrependReactor("create", "namespace", func(action ktesting.Action) (bool, runtime.Object, error) {
+				return true, nil, nil
+			})
+
+			fakeNamespace := testingutil.FakeNamespaceStatus(corev1.NamespacePhase("Active"), tt.projectName)
+			go func(project *corev1.Namespace) {
+				fkWatch.Add(project)
+			}(fakeNamespace)
+
+			fakeClientSet.Kubernetes.PrependWatchReactor("namespaces", func(action ktesting.Action) (handled bool, ret watch.Interface, err error) {
+				return true, fkWatch, nil
+			})
+
+			fkWatch2 := watch.NewFake()
+			go func() {
+				fkWatch2.Add(&corev1.ServiceAccount{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "default",
+					},
+				})
+			}()
+
+			fakeClientSet.Kubernetes.PrependWatchReactor("serviceaccounts", func(action ktesting.Action) (handled bool, ret watch.Interface, err error) {
+				return true, fkWatch2, nil
+			})
+
+			client.SetDiscoveryInterface(fakeDiscoveryWithNamespace)
+
+			// The function we are testing
+			err := Create(client, tt.projectName, true)
+
+			// Checks for error in positive cases
+			if !tt.wantErr == (err != nil) {
```

Comment:
can we reuse some parts from the above test e.g error checking, reactors etc?

Reply:
I tried that, but the problem is that almost every line is different due to Namespace vs Project calls

Reply:
OK